plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/custom/deserializer/AggregateMediatorExtFactory.java

```diff
@@ -31,7 +31,9 @@
 import org.apache.synapse.mediators.base.SequenceMediator;
 import org.apache.synapse.mediators.builtin.DropMediator;
 import org.apache.synapse.mediators.eip.aggregator.AggregateMediator;
+import org.apache.synapse.util.xpath.SynapseXPath;
 import org.jaxen.JaxenException;
+import org.wso2.developerstudio.eclipse.gmf.esb.internal.persistence.custom.SynapseXPathExt;
 
 public class AggregateMediatorExtFactory extends AggregateMediatorFactory {
```

Comment:
Shall we describe the functionality of this class? 

When coming from xml to diagram we need to validate using Synapse factory class (i.e AggregateMediatorFactory ). But as per exceptions we need a decorator class.

Reply:
Fixed ...